plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -406,6 +406,7 @@ export const createSessionReplayEvent = (
 ) => {
     const timestamps = events
         .filter((e) => !!e?.timestamp)
+        .filter((e) => Math.sign(e.timestamp) === 1)
         .map((e) => castTimestampOrNow(DateTime.fromMillis(e.timestamp), TimestampFormat.ClickHouse))
```

Comment:
```suggestion
        .filter((e) => Math.sign(e.timestamp) === 1)
        .map((e) => DateTime.fromMillis(e.timestamp))
        .filters((e) => e.isValid)
        .map((e) => castTimestampOrNow(e, TimestampFormat.ClickHouse))
```

(fly-by - I promise I'm not working 🙈)

we could add an explicit check?

https://github.com/moment/luxon/blob/master/docs/validity.md

Reply:
(core written in a GitHub comment, so definitely needs checking 🤣)